### ISSUE TITLE
[iOS and Mac] - Blank rendering in FlowDirection right to left from .net 7.0

### ISSUE THREAD

**Issue by RanjithkumarSyncfusion** [dotnet/maui#14406]
### Description

If the FlowDirection is RightToLeft and a ContentView is loaded inside a HorizontalStackLayout with a width request that is less than it, the ContentView's content is not presented in the view.

### Steps to Reproduce

I have attached the Test sample below
[ReproSample.zip](https://github.com/dotnet/maui/files/11158073/ReproSample.zip)

### Link to public reproduction project repository

-

### Version with bug

7.0 (current)

### Last version that worked well

6.0.424

### Affected platforms

iOS, macOS

### Affected platform versions

14.2

### Did you find any workaround?

https://github.com/dotnet/maui/blob/main/src/Core/src/Handlers/ViewHandlerExtensions.iOS.cs#LL100C48-L100

The method PlatformArrangeHandler uses the PlatformView.Bounds for RTL related conversion; this is OK for the initial call, but not if the same view is arranged more than once for the same X value. The number of arrange gets called the Bounds.X value will be (number of calls) n*CorrectValue.

**The correct approach will be using the Rect passed to PlatformArrangeHandler**

### Relevant log output

```shell
{{914, 0}, {50, 756}}
{{1828, 0}, {50, 756}}
{{2742, 0}, {50, 720}}
{{3656, 0}, {50, 720}}
{{4570, 0}, {50, 720}}
```

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with 17.6 Preview (17.6 build 1539). Can repro on iOS and macOS platform with sample project.
[ReproSample.zip](https://github.com/dotnet/maui/files/11720554/ReproSample.zip)
![Screenshot 2023-06-12 165422](https://github.com/dotnet/maui/assets/77434865/eb02af2c-d3d1-427c-9dff-0372390dadf6)
![Screenshot 2023-06-12 170048](https://github.com/dotnet/maui/assets/77434865/98058f4a-2fbf-4942-b605-6841a2e29dfa)

**Comment by hartez** [comment]
This is fixed by #15114.
